### PR TITLE
python3Packages.libnacl: 1.7.1 -> 1.7.2

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 1.7.2

Change log: not available (https://libnacl.readthedocs.io/en/latest/topics/releases/index.html#)

- switch to `pytestCheckHook`
- incl. `nixpkgs-fmt`

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 113050` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>python37Packages.asyncssh</li>
    <li>python37Packages.libnacl</li>
    <li>python38Packages.asyncssh</li>
    <li>python38Packages.libnacl</li>
    <li>python39Packages.asyncssh</li>
    <li>python39Packages.libnacl</li>
    <li>tribler</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113050` at e90ba9c0 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/e7d0b024ab42af312ae7e7b285981cb4">python37Packages.asyncssh</a></li></li>
    <li><a href="https://gist.github.com/c277385913c20f06a8f28c8118d2dc0e">python37Packages.libnacl</a></li></li>
    <li><a href="https://gist.github.com/ce6a69c4eab9dd87fa2e4d426bbd0efd">python38Packages.asyncssh</a></li></li>
    <li><a href="https://gist.github.com/0c86ac0c028ac1a9c8c7e870c90f8517">python38Packages.libnacl</a></li></li>
    <li><a href="https://gist.github.com/c7fa24a0a3f2748a3bd3f7a39d0ed80f">python39Packages.asyncssh</a></li></li>
    <li><a href="https://gist.github.com/cc30e1f9bdc6010633731b77636e7fa6">python39Packages.libnacl</a></li></li>
    <li><a href="https://gist.github.com/208dd031579098225c24e0b1b174314a">tribler</a></li></li>
  </ul>
</details>

File: pkgs/development/python-modules/libnacl/default.nix [hunk 1]
```diff
@@ -1,33 +1,39 @@
...
-  checkInputs = [ pytest ];
   propagatedBuildInputs = [ libsodium ];
 
   postPatch =
-    let soext = stdenv.hostPlatform.extensions.sharedLibrary; in ''
-    substituteInPlace "./libnacl/__init__.py" --replace "ctypes.cdll.LoadLibrary('libsodium${soext}')" "ctypes.cdll.LoadLibrary('${libsodium}/lib/libsodium${soext}')"
-  '';
+    let soext = stdenv.hostPlatform.extensions.sharedLibrary; in
+    ''
+      substituteInPlace "./libnacl/__init__.py" --replace "ctypes.cdll.LoadLibrary('libsodium${soext}')" "ctypes.cdll.LoadLibrary('${libsodium}/lib/libsodium${soext}')"
```
Comment (mweinelt) [inline pkgs/development/python-modules/libnacl/default.nix:25]: Please break this into multiple lines, 171 characters is a bit too much.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/libnacl/default.nix:25]:
```suggestion
      substituteInPlace "./libnacl/__init__.py" \
        --replace "ctypes.cdll.LoadLibrary('libsodium${soext}')" \ 
        "ctypes.cdll.LoadLibrary('${libsodium}/lib/libsodium${soext}')"
```

Comment (mweinelt) [reply pkgs/development/python-modules/libnacl/default.nix:25]:
```suggestion
      substituteInPlace "./libnacl/__init__.py" --replace \
        "ctypes.cdll.LoadLibrary('libsodium${soext}')" \ 
        "ctypes.cdll.LoadLibrary('${libsodium}/lib/libsodium${soext}')"
```
 
Enables better visual comparison of the replacement.

Comment (fabaff) [reply pkgs/development/python-modules/libnacl/default.nix]: Updated, thanks.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 113050` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>python37Packages.asyncssh</li>
    <li>python37Packages.libnacl</li>
    <li>python38Packages.asyncssh</li>
    <li>python38Packages.libnacl</li>
    <li>python39Packages.asyncssh</li>
    <li>python39Packages.libnacl</li>
    <li>tribler</li>
  </ul>
</details>

File: pkgs/development/python-modules/libnacl/default.nix [hunk 2]
```diff
@@ -1,33 +1,41 @@
...
   src = fetchFromGitHub {
     owner = "saltstack";
     repo = pname;
     rev = "v${version}";
-    sha256 = "10rpim9lf0qd861a3miq8iqg8w87slqwqni7nq66h72jdk130axg";
+    sha256 = "sha256-nttR9PQimhqd2pByJ5IJzJ4RmSI4y7lcX7a7jcK+vqc=";
   };
 
-  checkInputs = [ pytest ];
   propagatedBuildInputs = [ libsodium ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/libnacl/default.nix:20]: Shouldn't it sufficient for this to be in `buildInputs`?

Comment (fabaff) [reply pkgs/development/python-modules/libnacl/default.nix]: Yes, it's not needed during runtime.

```bash
ew/pr-113050-2/results/python38Packages.libnacl/lib/python3.8/site-packages]$ python
Python 3.7.9 (default, Aug 15 2020, 06:36:35) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import libnacl.public
>>> fred = libnacl.public.SecretKey()
>>> raw_sk = fred.sk
>>> print(raw_sk)
b'\x8a>\xfa\xfe4\xa7M \x06}D\xea@\xa4\x02M1w \xda\xb0L\x8f\x8dR\xc47\x01\x00\nxM'
```

**Comment by SuperSandro2000** [comment]
@fabaff you forgot to squash the commits.

**Comment by mweinelt** [comment]
@SuperSandro2000 The commits weren't signed, so I simply used GitHubs squash and merge feature.

**Pull request merged**